src/java.base/share/classes/java/text/CompactNumberFormat.java

```diff
@@ -589,6 +589,10 @@ private StringBuffer format(double number, StringBuffer result,
         if (compactDataIndex != -1) {
             long divisor = (Long) divisors.get(compactDataIndex);
             int iPart = getIntegerPart(number, divisor);
+            if (checkIncrement(iPart, compactDataIndex, divisor)) {
+                divisor = (Long) divisors.get(++compactDataIndex);
+                iPart = getIntegerPart(number, divisor);
+            }
```

Comment:
This and the few lines surrounding it were duplicated. Might be worth considering consolidation.

Reply:
Right. In fact, I tried to converge them but ended up not to, as there are slight differences for each number type.

Reply:
Yeah, that's fine.